draft-kohbrok-mimi-metadata-minimalization.md

```diff
@@ -65,95 +65,227 @@ required to achieve the goals detailed above.
 
-## Add flow
+At the time of registration with their local provider, Alice and Bob generate an
+HPKE keypair and upload the public key (called the connection establishment public key) to their respective providers. The providers make the keys
+available to all other users without requiring authentication.
 
-When adding a user(s) to a group via a Welcome message, the adding user must do
-two things: On the one hand, it must provide the new user(s) with the key
-material required to authenticate existing group members. On the other hand, it
-must provide the existing group members with the key material required to
-authenticate the new user(s).
+To hide Alice's identity from Bob's provider throughout the connection
+establishment process, Alice obtains Bob's connection establishment public key
+from Bob's provider.
 
-For the new user(s), the key material can be transported along with the Welcome,
-e.g. via a GroupInfo extension.
+Alice now creates a new conversation with her local provider as hub and adds
+all of her clients to that conversation. She then compiles the following
+information into a connection request:
 
-For existing users, the key material must be transported in some other way, e.g.
-via a custom proposal (encrypted under an exporter).
+~~~ tls
+struct {
+  Credential requester_client_credential;
+  opaque connection_room_id<V>;
+  opaque connection_room_identity_link_wrapper_key<V>;
+  opaque requester_connection_key<V>;
+  opaque connection_response_key<V>;
+} ConnectionRequest
+~~~
 
-## Join flow
+The `connection_response_key` is a fresh HPKE keypair that Alice generates when
+she creates the connection request. It is later used by Bob to encrypt his own
+connection key if Bob chooses to accept the request.
+
+TODO: The request will likely contain more information s.t. an authorization
+token that allows Bob to fetch Alice's KeyPackages, etc. If we want to include
+more information here, it should also be included in the ConnectionResponse
+below.
+
+Alice then encrypts the connection request using Bob's connection establishment
+public key and sends it to Bob's provider.
+
+Bob fetches the connection request from his provider and decrypts it using his connection establishment key.
+
+He can inspect Alice's `requester_client_credential` and decide whether to
+accept the request.
+
+If Bob accepts, he creates a connection response:
+
+~~~ tls
+struct {
+  opaque responder_connection_key<V>;
+} ConnectionResponse
+~~~
+
+Bob then follows the join flow for the room with the `connection_room_id`,
+including the ConnectionResponse encrypted under the `connection_response_key`
+in the join message's (i.e. the external commit's) AAD. As part of that
+process, he obtains the encrypted identity link keys for Alice's clients and
+verifies that they belong to the same user as the `requester_client_credential`
+in the connection request.
+
+Alice sees that Bob has accepted her connection request by joining the room and
```

Comment:
What does "sees" mean? Is it Alice the user or Alice the client?

Reply:
The event is detected by Alice the client and can be communicated to Alice the user. I'll clarify.